src/algebra/category/Module/basic.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2019 Robert A. Spencer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert A. Spencer
+-/
+import algebra.module
+import algebra.punit_instances
+import category_theory.concrete_category
+import linear_algebra.basic
+open category_theory
+
+universe u
+
+variables (R : Type u) [ring R]
+
+/-- The category of R-modules and their morphisms. -/
+structure Module :=
+(carrier : Type u)
+[is_add_comm_group : add_comm_group carrier]
+[is_module : module R carrier]
+
+attribute [instance] Module.is_add_comm_group Module.is_module
```

Comment:
Do these instances trigger on coerced `Module`s? If not, please restate them in terms of the coercion on line 27.